packages/input-option/src/InputOption/InputOption.tsx

```diff
@@ -3,57 +3,78 @@ import React from 'react';
 import { cx } from '@leafygreen-ui/emotion';
 import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
 import {
-  Polymorphic,
-  PolymorphicAs,
-  usePolymorphic,
+  InferredPolymorphic,
+  useInferredPolymorphic,
 } from '@leafygreen-ui/polymorphic';
 
 import {
-  inputOptionActiveStyles,
-  inputOptionDisabledStyles,
-  inputOptionHoverStyles,
+  disabledStyles,
+  getFormElementStyle,
+  getMenuElementStyle,
   inputOptionStyles,
-  inputOptionThemeStyles,
   inputOptionWedge,
-  titleSelectionStyles,
 } from './InputOption.style';
-import { InputOptionProps } from './InputOption.types';
+import {
+  ActionType,
+  InputOptionProps,
+  RenderedContext,
+} from './InputOption.types';
 
-export const InputOption = Polymorphic<InputOptionProps>(
+export const InputOption = InferredPolymorphic<InputOptionProps, 'div'>(
   (
     {
-      as = 'li' as PolymorphicAs,
+      as,
       children,
       disabled,
       highlighted,
-      selected,
+      checked,
       darkMode: darkModeProp,
       showWedge = true,
       isInteractive = true,
       className,
+      actionType = ActionType.Default,
+      renderedContext = RenderedContext.FormElement,
       ...rest
     },
     ref,
   ) => {
-    const { Component } = usePolymorphic(as);
+    const { Component } = useInferredPolymorphic(as, rest, 'div');
```

Comment:
Does this need to be inferred? Can we just leave it as basic polymorphic?

Reply:
Also, is defaulting to `div` intentional? Why not `li`?

Reply:
div is intentional - sometimes we'll want LI but not always. Divs are more flexible, more adaptable, and can be just as accessible so long as they receive the right aria attributes.

Reply:
I guess we could leave it as base polymorphic, and have the inferred logic happen in the components when they send it down?

Reply:
Yeah, I think stacking too many `Inferred` components will lead to headaches down the line